src/main/java/org/spongepowered/api/attributes/AttributeHolder.java

```diff
@@ -0,0 +1,80 @@
+package org.spongepowered.api.attributes;
+
+import com.google.common.base.Optional;
+
+import java.util.Collection;
+import java.util.List;
+
+public interface AttributeHolder {
+
+    /**
+     * Calculates the value of a certain {@link Attribute} on this holder.
+     * 
+     * @param attribute The Attribute to find the value of.
+     * @return The value of a certain Attribute on this holder, or
+     *         Optional.absent() if the attribute is not applicable.
+     */
+    Optional<Double> getAttributeValue(Attribute attribute);
+
+    /**
+     * Sets the base value of a certain {@link Attribute} on this holder.
+     * 
+     * @param attribute The Attribute to set the base value of.
+     * @param base The new base value.
+     * @throws IllegalArgumentException If the base value exceeds the maximum or
```

Comment:
Should mention this within the javadoc description, and possibly some link or reference to where to find the minimum/maximum allowed for the `Attribute`.